apps/mobile/src/utils/addToSortedArray.ts

```diff
@@ -16,7 +16,7 @@ export default function addToSortedArray<T>(
     if (index === -1) {
       array.push(item)
     } else {
-      array[index] = item
+      array.splice(index, 0, item)
```

Comment:
This is very good catch. Seems like this could bug the chat... Great job finding it